{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "523cf199",
   "metadata": {},
   "outputs": [],
   "source": [
    "import qiime2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "39a797c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "from qiime2 import Visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8efe4574",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Description: What is the influence of genotype (intrinsic) and environment (extrinsic) on anemone-associated bacterial communities?\n",
    "\n",
    "#Data: Illumina MiSeq v3 paired-end (2 × 300 bp) reads (FASTQ)\n",
    "\n",
    "#Tools: QIIME 2\n",
    "\n",
    "#Pipeline: \n",
    "#Section 1: Importing, cleaning and quality control of the data\n",
    "#Section 2: Taxonomic Analysis\n",
    "#Section 3: Building a phylogenetic tree\n",
    "#Section 4: Basic visualisations and statistics\n",
    "#Section 5: Exporting data for further analysis in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "264b1c2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#https://www.melbournebioinformatics.org.au/tutorials/tutorials/qiime2/qiime2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7d050e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Section 1: Importing, cleaning and quality control of the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dc84585",
   "metadata": {},
   "outputs": [],
   "source": [
    "#The raw data located in the directory raw_data is exported to a single QIIME 2 artefact file, combined.qza, located in the directory anaylsis/seqs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d7200e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p analysis/seqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fe6f85ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN10_Fwd01-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN10_Fwd01-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN10_Fwd01-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN10_Fwd01-Rev11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN10_Fwd01-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN10_Fwd01-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN10_Fwd01-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN10_Fwd01-Rev11_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        15,037 reads @  47.7 µs/read;   1.26 M reads/minute\n",
      "Finished in 0.731 s (48.622 µs/read; 1.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             15,037\n",
      "  Read 1 with adapter:                  15,033 (100.0%)\n",
      "  Read 2 with adapter:                  15,028 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        15,037 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,492,621 bp\n",
      "  Read 1:     4,403,058 bp\n",
      "  Read 2:     4,089,563 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      7,052,328 bp (83.0%)\n",
      "  Read 1:     3,652,876 bp\n",
      "  Read 2:     3,399,452 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 15033 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "44\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "46\t1\t0.0\t3\t1\n",
      "47\t18\t0.0\t3\t15 1 2\n",
      "48\t151\t0.0\t3\t106 30 15\n",
      "49\t1122\t0.0\t3\t802 310 9 1\n",
      "50\t13715\t0.0\t3\t13257 447 11\n",
      "51\t23\t0.0\t3\t9 14\n",
      "52\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 15028 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "42\t3\t0.0\t3\t3\n",
      "43\t9\t0.0\t3\t9\n",
      "44\t99\t0.0\t3\t61 19 16 3\n",
      "45\t955\t0.0\t3\t603 328 22 2\n",
      "46\t13947\t0.0\t3\t12740 1173 33 1\n",
      "47\t15\t0.0\t3\t10 5\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN22_Fwd02-Rev07_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN22_Fwd02-Rev07_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN22_Fwd02-Rev07_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN22_Fwd02-Rev07_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN22_Fwd02-Rev07_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN22_Fwd02-Rev07_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN22_Fwd02-Rev07_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN22_Fwd02-Rev07_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        15,779 reads @  46.3 µs/read;   1.29 M reads/minute\n",
      "Finished in 0.741 s (46.983 µs/read; 1.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             15,779\n",
      "  Read 1 with adapter:                  15,773 (100.0%)\n",
      "  Read 2 with adapter:                  15,759 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs written (passing filters):        15,778 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,921,169 bp\n",
      "  Read 1:     4,632,093 bp\n",
      "  Read 2:     4,289,076 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      7,347,475 bp (82.4%)\n",
      "  Read 1:     3,829,252 bp\n",
      "  Read 2:     3,518,223 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 15773 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "43\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "46\t3\t0.0\t3\t3\n",
      "47\t9\t0.0\t3\t7 1 0 1\n",
      "48\t29\t0.0\t3\t23 3 2 1\n",
      "49\t170\t0.0\t3\t132 23 15\n",
      "50\t1376\t0.0\t3\t1042 328 6\n",
      "51\t14152\t0.0\t3\t13664 478 9 1\n",
      "52\t27\t0.0\t3\t12 14 1\n",
      "53\t5\t0.0\t3\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 15759 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "37\t1\t0.0\t3\t0 0 0 1\n",
      "43\t3\t0.0\t3\t1 1 0 1\n",
      "44\t3\t0.0\t3\t1 2\n",
      "45\t4\t0.0\t3\t1 3\n",
      "46\t20\t0.0\t3\t12 5 1 2\n",
      "47\t99\t0.0\t3\t62 17 19 1\n",
      "48\t1075\t0.0\t3\t729 332 12 2\n",
      "49\t14513\t0.0\t3\t14047 436 20 10\n",
      "50\t26\t0.0\t3\t12 14\n",
      "51\t15\t0.0\t3\t15\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN27_Fwd02-Rev13_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN27_Fwd02-Rev13_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN27_Fwd02-Rev13_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN27_Fwd02-Rev13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN27_Fwd02-Rev13_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN27_Fwd02-Rev13_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN27_Fwd02-Rev13_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN27_Fwd02-Rev13_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done           00:00:00        13,612 reads @  47.2 µs/read;   1.27 M reads/minute\n",
      "Finished in 0.650 s (47.770 µs/read; 1.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             13,612\n",
      "  Read 1 with adapter:                  13,609 (100.0%)\n",
      "  Read 2 with adapter:                  13,597 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        13,612 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,704,804 bp\n",
      "  Read 1:     4,017,738 bp\n",
      "  Read 2:     3,687,066 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      6,374,606 bp (82.7%)\n",
      "  Read 1:     3,325,259 bp\n",
      "  Read 2:     3,049,347 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 13609 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "46\t2\t0.0\t3\t1 0 0 1\n",
      "47\t6\t0.0\t3\t6\n",
      "48\t25\t0.0\t3\t19 3 2 1\n",
      "49\t158\t0.0\t3\t114 26 17 1\n",
      "50\t1176\t0.0\t3\t864 305 6 1\n",
      "51\t12210\t0.0\t3\t11778 421 11\n",
      "52\t21\t0.0\t3\t10 10 1\n",
      "53\t9\t0.0\t3\t9\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 13597 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "41\t1\t0.0\t3\t0 0 1\n",
      "42\t1\t0.0\t3\t1\n",
      "43\t2\t0.0\t3\t2\n",
      "44\t22\t0.0\t3\t12 5 3 2\n",
      "45\t110\t0.0\t3\t67 23 19 1\n",
      "46\t1057\t0.0\t3\t726 322 9\n",
      "47\t12382\t0.0\t3\t11987 364 27 4\n",
      "48\t22\t0.0\t3\t7 14 1\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN40_Fwd03-Rev04_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN40_Fwd03-Rev04_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN40_Fwd03-Rev04_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN40_Fwd03-Rev04_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN40_Fwd03-Rev04_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN40_Fwd03-Rev04_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN40_Fwd03-Rev04_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN40_Fwd03-Rev04_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        14,611 reads @  47.2 µs/read;   1.27 M reads/minute\n",
      "Finished in 0.699 s (47.842 µs/read; 1.25 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,611\n",
      "  Read 1 with adapter:                  14,603 (99.9%)\n",
      "  Read 2 with adapter:                  14,597 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        14,611 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,201,693 bp\n",
      "  Read 1:     4,238,121 bp\n",
      "  Read 2:     3,963,572 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      6,773,588 bp (82.6%)\n",
      "  Read 1:     3,480,226 bp\n",
      "  Read 2:     3,293,362 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 14603 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "42\t1\t0.0\t3\t1\n",
      "43\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "47\t2\t0.0\t3\t2\n",
      "48\t5\t0.0\t3\t5\n",
      "49\t18\t0.0\t3\t16 2\n",
      "50\t119\t0.0\t3\t85 21 13\n",
      "51\t1135\t0.0\t3\t823 293 16 3\n",
      "52\t13299\t0.0\t3\t12827 450 19 3\n",
      "53\t22\t0.0\t3\t9 13\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 14597 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "39\t1\t0.0\t3\t1\n",
      "41\t2\t0.0\t3\t2\n",
      "42\t3\t0.0\t3\t2 0 1\n",
      "43\t15\t0.0\t3\t8 3 2 2\n",
      "44\t96\t0.0\t3\t57 23 15 1\n",
      "45\t1007\t0.0\t3\t646 341 20\n",
      "46\t13454\t0.0\t3\t12349 1074 29 2\n",
      "47\t18\t0.0\t3\t9 6 2 1\n",
      "49\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN45_Fwd03-Rev09_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN45_Fwd03-Rev09_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN45_Fwd03-Rev09_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN45_Fwd03-Rev09_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN45_Fwd03-Rev09_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN45_Fwd03-Rev09_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN45_Fwd03-Rev09_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN45_Fwd03-Rev09_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        18,187 reads @  48.5 µs/read;   1.24 M reads/minute\n",
      "Finished in 0.894 s (49.179 µs/read; 1.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             18,187\n",
      "  Read 1 with adapter:                  18,179 (100.0%)\n",
      "  Read 2 with adapter:                  18,175 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        18,187 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,199,573 bp\n",
      "  Read 1:     5,281,703 bp\n",
      "  Read 2:     4,917,870 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      8,421,508 bp (82.6%)\n",
      "  Read 1:     4,338,153 bp\n",
      "  Read 2:     4,083,355 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 18179 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "44\t2\t0.0\t3\t1 1\n",
      "45\t1\t0.0\t3\t1\n",
      "47\t2\t0.0\t3\t2\n",
      "48\t6\t0.0\t3\t5 1\n",
      "49\t21\t0.0\t3\t15 2 2 2\n",
      "50\t167\t0.0\t3\t120 31 15 1\n",
      "51\t1337\t0.0\t3\t965 360 9 3\n",
      "52\t16610\t0.0\t3\t15951 627 26 6\n",
      "53\t33\t0.0\t3\t12 20 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 18175 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t1\n",
      "41\t1\t0.0\t3\t1\n",
      "42\t8\t0.0\t3\t6 1 0 1\n",
      "43\t15\t0.0\t3\t10 3 1 1\n",
      "44\t126\t0.0\t3\t77 26 22 1\n",
      "45\t1213\t0.0\t3\t732 451 28 2\n",
      "46\t16793\t0.0\t3\t15174 1570 45 4\n",
      "47\t18\t0.0\t3\t8 9 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN48_Fwd03-Rev13_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN48_Fwd03-Rev13_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN48_Fwd03-Rev13_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN48_Fwd03-Rev13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN48_Fwd03-Rev13_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN48_Fwd03-Rev13_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN48_Fwd03-Rev13_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN48_Fwd03-Rev13_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        14,083 reads @  47.1 µs/read;   1.27 M reads/minute\n",
      "Finished in 0.676 s (48.026 µs/read; 1.25 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,083\n",
      "  Read 1 with adapter:                  14,082 (100.0%)\n",
      "  Read 2 with adapter:                  14,078 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        14,083 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,917,605 bp\n",
      "  Read 1:     4,083,833 bp\n",
      "  Read 2:     3,833,772 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      6,526,471 bp (82.4%)\n",
      "  Read 1:     3,352,980 bp\n",
      "  Read 2:     3,173,491 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 14082 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "43\t1\t0.0\t3\t0 1\n",
      "46\t1\t0.0\t3\t1\n",
      "47\t1\t0.0\t3\t1\n",
      "48\t4\t0.0\t3\t2 1 1\n",
      "49\t22\t0.0\t3\t20 1 0 1\n",
      "50\t139\t0.0\t3\t91 24 24\n",
      "51\t1056\t0.0\t3\t758 285 12 1\n",
      "52\t12833\t0.0\t3\t12368 441 20 4\n",
      "53\t25\t0.0\t3\t17 8\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 14078 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "39\t1\t0.0\t3\t1\n",
      "43\t5\t0.0\t3\t3 2\n",
      "44\t16\t0.0\t3\t13 0 1 2\n",
      "45\t91\t0.0\t3\t51 26 14\n",
      "46\t1143\t0.0\t3\t803 334 5 1\n",
      "47\t12806\t0.0\t3\t12464 329 11 2\n",
      "48\t16\t0.0\t3\t7 9\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN58_Fwd04-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN58_Fwd04-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN58_Fwd04-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN58_Fwd04-Rev11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN58_Fwd04-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN58_Fwd04-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN58_Fwd04-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN58_Fwd04-Rev11_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        11,780 reads @  56.2 µs/read;   1.07 M reads/minute\n",
      "Finished in 0.668 s (56.740 µs/read; 1.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             11,780\n",
      "  Read 1 with adapter:                  11,775 (100.0%)\n",
      "  Read 2 with adapter:                  11,772 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        11,780 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,693,438 bp\n",
      "  Read 1:     3,490,418 bp\n",
      "  Read 2:     3,203,020 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      5,577,039 bp (83.3%)\n",
      "  Read 1:     2,914,555 bp\n",
      "  Read 2:     2,662,484 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 11775 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "42\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "46\t8\t0.0\t3\t7 0 0 1\n",
      "47\t94\t0.0\t3\t71 14 9\n",
      "48\t900\t0.0\t3\t632 259 8 1\n",
      "49\t10754\t0.0\t3\t10438 309 6 1\n",
      "50\t16\t0.0\t3\t6 10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 11772 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t1\n",
      "41\t2\t0.0\t3\t0 1 1\n",
      "42\t1\t0.0\t3\t1\n",
      "43\t19\t0.0\t3\t12 3 2 2\n",
      "44\t72\t0.0\t3\t43 15 14\n",
      "45\t778\t0.0\t3\t469 284 23 2\n",
      "46\t10876\t0.0\t3\t9887 965 20 4\n",
      "47\t23\t0.0\t3\t9 11 3\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN66_Fwd05-Rev06_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN66_Fwd05-Rev06_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN66_Fwd05-Rev06_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN66_Fwd05-Rev06_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN66_Fwd05-Rev06_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN66_Fwd05-Rev06_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN66_Fwd05-Rev06_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN66_Fwd05-Rev06_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        10,771 reads @  49.4 µs/read;   1.22 M reads/minute\n",
      "Finished in 0.539 s (50.068 µs/read; 1.20 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             10,771\n",
      "  Read 1 with adapter:                  10,771 (100.0%)\n",
      "  Read 2 with adapter:                  10,766 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        10,771 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,130,130 bp\n",
      "  Read 1:     3,201,175 bp\n",
      "  Read 2:     2,928,955 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      5,098,656 bp (83.2%)\n",
      "  Read 1:     2,663,936 bp\n",
      "  Read 2:     2,434,720 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 10771 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t168.3\t0\t6\n",
      "45\t2\t0.0\t3\t1 0 0 1\n",
      "46\t2\t0.0\t3\t2\n",
      "47\t8\t0.0\t3\t6 1 0 1\n",
      "48\t92\t0.0\t3\t68 9 15\n",
      "49\t812\t0.0\t3\t602 207 3\n",
      "50\t9840\t0.0\t3\t9578 253 9\n",
      "51\t9\t0.0\t3\t6 3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 10766 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "32\t1\t0.0\t3\t0 1\n",
      "42\t4\t0.0\t3\t2 2\n",
      "43\t12\t0.0\t3\t8 3 1\n",
      "44\t71\t0.0\t3\t45 14 10 2\n",
      "45\t804\t0.0\t3\t513 268 23\n",
      "46\t9863\t0.0\t3\t8909 917 32 5\n",
      "47\t11\t0.0\t3\t6 5\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN67_Fwd05-Rev07_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN67_Fwd05-Rev07_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN67_Fwd05-Rev07_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN67_Fwd05-Rev07_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN67_Fwd05-Rev07_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN67_Fwd05-Rev07_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN67_Fwd05-Rev07_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN67_Fwd05-Rev07_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        12,855 reads @  45.3 µs/read;   1.32 M reads/minute\n",
      "Finished in 0.589 s (45.843 µs/read; 1.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             12,855\n",
      "  Read 1 with adapter:                  12,855 (100.0%)\n",
      "  Read 2 with adapter:                  12,844 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        12,855 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,323,659 bp\n",
      "  Read 1:     3,823,404 bp\n",
      "  Read 2:     3,500,255 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      6,054,113 bp (82.7%)\n",
      "  Read 1:     3,182,141 bp\n",
      "  Read 2:     2,871,972 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 12855 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t200.9\t0\t4\n",
      "36\t1\t0.0\t3\t1\n",
      "45\t4\t0.0\t3\t3 1\n",
      "46\t3\t0.0\t3\t2 0 1\n",
      "47\t18\t0.0\t3\t14 2 1 1\n",
      "48\t115\t0.0\t3\t88 18 7 2\n",
      "49\t1000\t0.0\t3\t724 273 3\n",
      "50\t11690\t0.0\t3\t11396 286 7 1\n",
      "51\t19\t0.0\t3\t8 11\n",
      "62\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 12844 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "42\t2\t0.0\t3\t1 0 0 1\n",
      "44\t3\t0.0\t3\t3\n",
      "45\t4\t0.0\t3\t4\n",
      "46\t19\t0.0\t3\t10 8 1\n",
      "47\t92\t0.0\t3\t58 19 15\n",
      "48\t845\t0.0\t3\t565 276 4\n",
      "49\t11835\t0.0\t3\t11493 319 17 6\n",
      "50\t30\t0.0\t3\t13 16 1\n",
      "51\t14\t0.0\t3\t14\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN70_Fwd05-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN70_Fwd05-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN70_Fwd05-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN70_Fwd05-Rev11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN70_Fwd05-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN70_Fwd05-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN70_Fwd05-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN70_Fwd05-Rev11_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        14,758 reads @  46.1 µs/read;   1.30 M reads/minute\n",
      "Finished in 0.692 s (46.912 µs/read; 1.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,758\n",
      "  Read 1 with adapter:                  14,758 (100.0%)\n",
      "  Read 2 with adapter:                  14,751 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        14,758 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,396,339 bp\n",
      "  Read 1:     4,391,100 bp\n",
      "  Read 2:     4,005,239 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      6,982,763 bp (83.2%)\n",
      "  Read 1:     3,654,764 bp\n",
      "  Read 2:     3,327,999 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 14758 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t230.6\t0\t4\n",
      "42\t1\t0.0\t3\t1\n",
      "43\t1\t0.0\t3\t1\n",
      "46\t2\t0.0\t3\t2\n",
      "47\t9\t0.0\t3\t8 1\n",
      "48\t119\t0.0\t3\t85 22 12\n",
      "49\t1111\t0.0\t3\t814 289 8\n",
      "50\t13488\t0.0\t3\t13126 352 10\n",
      "51\t23\t0.0\t3\t13 9 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 14751 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t2\t0.0\t3\t2\n",
      "42\t5\t0.0\t3\t2 0 3\n",
      "43\t16\t0.0\t3\t8 2 4 2\n",
      "44\t103\t0.0\t3\t59 19 22 3\n",
      "45\t1047\t0.0\t3\t655 358 32 2\n",
      "46\t13551\t0.0\t3\t12252 1253 42 4\n",
      "47\t27\t0.0\t3\t11 14 2\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN73_Fwd06-Rev01_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN73_Fwd06-Rev01_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN73_Fwd06-Rev01_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN73_Fwd06-Rev01_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN73_Fwd06-Rev01_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN73_Fwd06-Rev01_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN73_Fwd06-Rev01_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN73_Fwd06-Rev01_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:01        16,806 reads @  67.8 µs/read;   0.88 M reads/minute\n",
      "Finished in 1.150 s (68.433 µs/read; 0.88 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             16,806\n",
      "  Read 1 with adapter:                  16,803 (100.0%)\n",
      "  Read 2 with adapter:                  16,791 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        16,806 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,535,637 bp\n",
      "  Read 1:     4,989,031 bp\n",
      "  Read 2:     4,546,606 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      7,926,246 bp (83.1%)\n",
      "  Read 1:     4,167,349 bp\n",
      "  Read 2:     3,758,897 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 16803 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "44\t2\t0.0\t3\t1 0 0 1\n",
      "45\t8\t0.0\t3\t7 0 0 1\n",
      "46\t21\t0.0\t3\t17 1 2 1\n",
      "47\t119\t0.0\t3\t88 19 12\n",
      "48\t1344\t0.0\t3\t962 374 7 1\n",
      "49\t15287\t0.0\t3\t14810 458 16 3\n",
      "50\t22\t0.0\t3\t13 9\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 16791 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "39\t1\t0.0\t3\t1\n",
      "41\t3\t0.0\t3\t1 0 2\n",
      "42\t2\t0.0\t3\t0 1 1\n",
      "43\t2\t0.0\t3\t2\n",
      "44\t17\t0.0\t3\t11 3 2 1\n",
      "45\t111\t0.0\t3\t75 22 13 1\n",
      "46\t1205\t0.0\t3\t778 417 10\n",
      "47\t15432\t0.0\t3\t14979 415 31 7\n",
      "48\t17\t0.0\t3\t9 8\n",
      "84\t1\t0.0\t3\t1\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN74_Fwd06-Rev02_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN74_Fwd06-Rev02_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN74_Fwd06-Rev02_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN74_Fwd06-Rev02_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN74_Fwd06-Rev02_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN74_Fwd06-Rev02_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN74_Fwd06-Rev02_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN74_Fwd06-Rev02_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        12,482 reads @  50.5 µs/read;   1.19 M reads/minute\n",
      "Finished in 0.642 s (51.470 µs/read; 1.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             12,482\n",
      "  Read 1 with adapter:                  12,479 (100.0%)\n",
      "  Read 2 with adapter:                  12,471 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        12,482 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,060,330 bp\n",
      "  Read 1:     3,693,930 bp\n",
      "  Read 2:     3,366,400 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      5,852,590 bp (82.9%)\n",
      "  Read 1:     3,083,664 bp\n",
      "  Read 2:     2,768,926 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 12479 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "35\t1\t0.0\t3\t1\n",
      "45\t5\t0.0\t3\t4 0 1\n",
      "46\t14\t0.0\t3\t11 1 0 2\n",
      "47\t114\t0.0\t3\t83 21 10\n",
      "48\t919\t0.0\t3\t656 251 12\n",
      "49\t11409\t0.0\t3\t11052 345 12\n",
      "50\t16\t0.0\t3\t9 7\n",
      "51\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 12471 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "42\t1\t0.0\t3\t0 0 0 1\n",
      "43\t3\t0.0\t3\t1 2\n",
      "44\t3\t0.0\t3\t1 1 0 1\n",
      "45\t12\t0.0\t3\t9 2 1\n",
      "46\t104\t0.0\t3\t58 34 12\n",
      "47\t877\t0.0\t3\t563 306 7 1\n",
      "48\t11451\t0.0\t3\t11080 347 16 8\n",
      "49\t20\t0.0\t3\t6 14\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN76_Fwd06-Rev04_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN76_Fwd06-Rev04_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN76_Fwd06-Rev04_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN76_Fwd06-Rev04_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN76_Fwd06-Rev04_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN76_Fwd06-Rev04_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN76_Fwd06-Rev04_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN76_Fwd06-Rev04_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        16,818 reads @  49.2 µs/read;   1.22 M reads/minute\n",
      "Finished in 0.835 s (49.664 µs/read; 1.21 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             16,818\n",
      "  Read 1 with adapter:                  16,816 (100.0%)\n",
      "  Read 2 with adapter:                  16,811 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        16,818 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,537,344 bp\n",
      "  Read 1:     4,991,017 bp\n",
      "  Read 2:     4,546,327 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      7,943,168 bp (83.3%)\n",
      "  Read 1:     4,168,651 bp\n",
      "  Read 2:     3,774,517 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 16816 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "44\t3\t0.0\t3\t2 0 0 1\n",
      "45\t3\t0.0\t3\t2 0 1\n",
      "46\t22\t0.0\t3\t17 1 3 1\n",
      "47\t144\t0.0\t3\t98 28 18\n",
      "48\t1259\t0.0\t3\t868 378 10 3\n",
      "49\t15364\t0.0\t3\t14901 451 9 3\n",
      "50\t20\t0.0\t3\t7 13\n",
      "51\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 16811 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "38\t1\t0.0\t3\t0 1\n",
      "42\t4\t0.0\t3\t4\n",
      "43\t12\t0.0\t3\t10 0 2\n",
      "44\t131\t0.0\t3\t71 28 30 2\n",
      "45\t1185\t0.0\t3\t786 368 28 3\n",
      "46\t15467\t0.0\t3\t14004 1413 39 11\n",
      "47\t11\t0.0\t3\t5 4 2\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN77_Fwd06-Rev05_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN77_Fwd06-Rev05_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN77_Fwd06-Rev05_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN77_Fwd06-Rev05_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN77_Fwd06-Rev05_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN77_Fwd06-Rev05_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN77_Fwd06-Rev05_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN77_Fwd06-Rev05_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:01        18,675 reads @  56.6 µs/read;   1.06 M reads/minute\n",
      "Finished in 1.065 s (57.040 µs/read; 1.05 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             18,675\n",
      "  Read 1 with adapter:                  18,672 (100.0%)\n",
      "  Read 2 with adapter:                  18,663 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        18,675 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,611,252 bp\n",
      "  Read 1:     5,543,297 bp\n",
      "  Read 2:     5,067,955 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      8,822,780 bp (83.1%)\n",
      "  Read 1:     4,630,216 bp\n",
      "  Read 2:     4,192,564 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 18672 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t1\n",
      "44\t2\t0.0\t3\t2\n",
      "45\t3\t0.0\t3\t3\n",
      "46\t26\t0.0\t3\t18 5 3\n",
      "47\t156\t0.0\t3\t117 24 15\n",
      "48\t1462\t0.0\t3\t1039 412 9 2\n",
      "49\t16986\t0.0\t3\t16493 480 13\n",
      "50\t36\t0.0\t3\t22 14\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 18663 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "36\t1\t0.0\t3\t0 0 0 1\n",
      "37\t1\t0.0\t3\t1\n",
      "41\t1\t0.0\t3\t1\n",
      "42\t2\t0.0\t3\t2\n",
      "43\t3\t0.0\t3\t2 0 1\n",
      "44\t22\t0.0\t3\t15 4 0 3\n",
      "45\t148\t0.0\t3\t90 33 24 1\n",
      "46\t1378\t0.0\t3\t910 453 14 1\n",
      "47\t17088\t0.0\t3\t16531 523 25 9\n",
      "48\t19\t0.0\t3\t4 15\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN78_Fwd06-Rev06_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN78_Fwd06-Rev06_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN78_Fwd06-Rev06_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN78_Fwd06-Rev06_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN78_Fwd06-Rev06_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN78_Fwd06-Rev06_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN78_Fwd06-Rev06_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN78_Fwd06-Rev06_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        16,591 reads @  57.7 µs/read;   1.04 M reads/minute\n",
      "Finished in 0.967 s (58.311 µs/read; 1.03 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             16,591\n",
      "  Read 1 with adapter:                  16,583 (100.0%)\n",
      "  Read 2 with adapter:                  16,581 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        16,591 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,402,592 bp\n",
      "  Read 1:     4,923,673 bp\n",
      "  Read 2:     4,478,919 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      7,830,335 bp (83.3%)\n",
      "  Read 1:     4,112,703 bp\n",
      "  Read 2:     3,717,632 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 16583 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "41\t1\t0.0\t3\t1\n",
      "43\t1\t0.0\t3\t0 1\n",
      "44\t3\t0.0\t3\t3\n",
      "45\t6\t0.0\t3\t4 2\n",
      "46\t17\t0.0\t3\t13 3 1\n",
      "47\t124\t0.0\t3\t91 19 14\n",
      "48\t1271\t0.0\t3\t898 362 11\n",
      "49\t15135\t0.0\t3\t14697 427 10 1\n",
      "50\t24\t0.0\t3\t6 18\n",
      "51\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 16581 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t0 0 1\n",
      "42\t3\t0.0\t3\t3\n",
      "43\t14\t0.0\t3\t8 3 3\n",
      "44\t111\t0.0\t3\t64 30 17\n",
      "45\t1179\t0.0\t3\t783 377 17 2\n",
      "46\t15251\t0.0\t3\t13855 1346 47 3\n",
      "47\t22\t0.0\t3\t8 11 3\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN91_Fwd07-Rev07_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN91_Fwd07-Rev07_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN91_Fwd07-Rev07_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN91_Fwd07-Rev07_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN91_Fwd07-Rev07_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN91_Fwd07-Rev07_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN91_Fwd07-Rev07_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN91_Fwd07-Rev07_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        10,280 reads @  54.7 µs/read;   1.10 M reads/minute\n",
      "Finished in 0.573 s (55.714 µs/read; 1.08 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             10,280\n",
      "  Read 1 with adapter:                  10,276 (100.0%)\n",
      "  Read 2 with adapter:                  10,272 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        10,280 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,843,274 bp\n",
      "  Read 1:     3,039,667 bp\n",
      "  Read 2:     2,803,607 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      4,807,593 bp (82.3%)\n",
      "  Read 1:     2,506,443 bp\n",
      "  Read 2:     2,301,150 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 10276 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "45\t1\t0.0\t3\t1\n",
      "46\t1\t0.0\t3\t1\n",
      "47\t4\t0.0\t3\t4\n",
      "48\t7\t0.0\t3\t5 2\n",
      "49\t20\t0.0\t3\t17 1 0 2\n",
      "50\t105\t0.0\t3\t81 22 2\n",
      "51\t814\t0.0\t3\t567 240 7\n",
      "52\t9307\t0.0\t3\t9005 296 5 1\n",
      "53\t17\t0.0\t3\t10 7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 10272 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "43\t1\t0.0\t3\t0 0 1\n",
      "45\t3\t0.0\t3\t3\n",
      "46\t8\t0.0\t3\t4 1 2 1\n",
      "47\t82\t0.0\t3\t50 15 17\n",
      "48\t704\t0.0\t3\t449 239 14 2\n",
      "49\t9445\t0.0\t3\t9128 304 9 4\n",
      "50\t19\t0.0\t3\t14 5\n",
      "51\t10\t0.0\t3\t9 1\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN92_Fwd07-Rev08_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN92_Fwd07-Rev08_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN92_Fwd07-Rev08_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN92_Fwd07-Rev08_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN92_Fwd07-Rev08_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN92_Fwd07-Rev08_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN92_Fwd07-Rev08_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN92_Fwd07-Rev08_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00         8,874 reads @  50.3 µs/read;   1.19 M reads/minute\n",
      "Finished in 0.458 s (51.596 µs/read; 1.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              8,874\n",
      "  Read 1 with adapter:                   8,871 (100.0%)\n",
      "  Read 2 with adapter:                   8,863 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):         8,874 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,041,627 bp\n",
      "  Read 1:     2,622,834 bp\n",
      "  Read 2:     2,418,793 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      4,156,708 bp (82.4%)\n",
      "  Read 1:     2,162,592 bp\n",
      "  Read 2:     1,994,116 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 8871 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "41\t1\t0.0\t3\t1\n",
      "46\t2\t0.0\t3\t2\n",
      "47\t3\t0.0\t3\t3\n",
      "48\t4\t0.0\t3\t2 0 1 1\n",
      "49\t16\t0.0\t3\t12 2 1 1\n",
      "50\t105\t0.0\t3\t77 18 10\n",
      "51\t749\t0.0\t3\t543 200 5 1\n",
      "52\t7980\t0.0\t3\t7746 225 6 3\n",
      "53\t11\t0.0\t3\t8 3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 8863 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t1\n",
      "43\t1\t0.0\t3\t1\n",
      "45\t4\t0.0\t3\t2 2\n",
      "46\t48\t0.0\t3\t25 12 11\n",
      "47\t639\t0.0\t3\t413 225 1\n",
      "48\t8157\t0.0\t3\t7911 227 18 1\n",
      "49\t13\t0.0\t3\t8 4 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN93_Fwd07-Rev09_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN93_Fwd07-Rev09_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN93_Fwd07-Rev09_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN93_Fwd07-Rev09_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN93_Fwd07-Rev09_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN93_Fwd07-Rev09_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN93_Fwd07-Rev09_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN93_Fwd07-Rev09_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        12,122 reads @  60.8 µs/read;   0.99 M reads/minute\n",
      "Finished in 0.748 s (61.704 µs/read; 0.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             12,122\n",
      "  Read 1 with adapter:                  12,120 (100.0%)\n",
      "  Read 2 with adapter:                  12,118 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        12,122 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,870,202 bp\n",
      "  Read 1:     3,588,049 bp\n",
      "  Read 2:     3,282,153 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      5,684,827 bp (82.7%)\n",
      "  Read 1:     2,959,054 bp\n",
      "  Read 2:     2,725,773 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 12120 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "40\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "47\t4\t0.0\t3\t4\n",
      "48\t2\t0.0\t3\t2\n",
      "49\t9\t0.0\t3\t5 2 2\n",
      "50\t100\t0.0\t3\t75 22 3\n",
      "51\t987\t0.0\t3\t706 276 4 1\n",
      "52\t11000\t0.0\t3\t10704 291 5\n",
      "53\t16\t0.0\t3\t11 5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 12118 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "41\t3\t0.0\t3\t2 1\n",
      "42\t3\t0.0\t3\t3\n",
      "43\t9\t0.0\t3\t3 3 2 1\n",
      "44\t82\t0.0\t3\t54 16 11 1\n",
      "45\t848\t0.0\t3\t539 285 22 2\n",
      "46\t11156\t0.0\t3\t10093 1028 27 8\n",
      "47\t16\t0.0\t3\t6 10\n",
      "48\t1\t0.0\t3\t1\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN94_Fwd07-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN94_Fwd07-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN94_Fwd07-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN94_Fwd07-Rev11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN94_Fwd07-Rev11_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN94_Fwd07-Rev11_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN94_Fwd07-Rev11_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN94_Fwd07-Rev11_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        10,511 reads @  55.4 µs/read;   1.08 M reads/minute\n",
      "Finished in 0.595 s (56.635 µs/read; 1.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             10,511\n",
      "  Read 1 with adapter:                  10,507 (100.0%)\n",
      "  Read 2 with adapter:                  10,508 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        10,511 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,962,516 bp\n",
      "  Read 1:     3,105,377 bp\n",
      "  Read 2:     2,857,139 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      4,934,993 bp (82.8%)\n",
      "  Read 1:     2,560,224 bp\n",
      "  Read 2:     2,374,769 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 10507 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "41\t1\t0.0\t3\t1\n",
      "47\t2\t0.0\t3\t2\n",
      "48\t6\t0.0\t3\t5 0 1\n",
      "49\t26\t0.0\t3\t22 2 2\n",
      "50\t106\t0.0\t3\t79 16 11\n",
      "51\t890\t0.0\t3\t642 242 6\n",
      "52\t9462\t0.0\t3\t9173 280 8 1\n",
      "53\t14\t0.0\t3\t6 8\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 10508 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t1\t0.0\t3\t0 0 0 1\n",
      "42\t4\t0.0\t3\t3 0 1\n",
      "43\t12\t0.0\t3\t7 2 3\n",
      "44\t88\t0.0\t3\t58 20 8 2\n",
      "45\t758\t0.0\t3\t481 264 13\n",
      "46\t9631\t0.0\t3\t8725 881 23 2\n",
      "47\t14\t0.0\t3\t9 5\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN95_Fwd07-Rev12_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN95_Fwd07-Rev12_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN95_Fwd07-Rev12_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN95_Fwd07-Rev12_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN95_Fwd07-Rev12_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/AN95_Fwd07-Rev12_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN95_Fwd07-Rev12_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/AN95_Fwd07-Rev12_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00         9,157 reads @  52.3 µs/read;   1.15 M reads/minute\n",
      "Finished in 0.490 s (53.535 µs/read; 1.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              9,157\n",
      "  Read 1 with adapter:                   9,154 (100.0%)\n",
      "  Read 2 with adapter:                   9,152 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):         9,157 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,202,962 bp\n",
      "  Read 1:     2,707,344 bp\n",
      "  Read 2:     2,495,618 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      4,289,366 bp (82.4%)\n",
      "  Read 1:     2,232,366 bp\n",
      "  Read 2:     2,057,000 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 9154 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "42\t1\t0.0\t3\t1\n",
      "46\t1\t0.0\t3\t1\n",
      "47\t2\t0.0\t3\t1 0 0 1\n",
      "48\t1\t0.0\t3\t1\n",
      "49\t16\t0.0\t3\t13 3\n",
      "50\t92\t0.0\t3\t64 16 11 1\n",
      "51\t780\t0.0\t3\t570 202 8\n",
      "52\t8249\t0.0\t3\t7984 263 2\n",
      "53\t12\t0.0\t3\t7 4 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 9152 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "44\t1\t0.0\t3\t0 0 0 1\n",
      "45\t11\t0.0\t3\t3 4 0 4\n",
      "46\t38\t0.0\t3\t14 17 7\n",
      "47\t578\t0.0\t3\t358 211 7 2\n",
      "48\t8511\t0.0\t3\t8258 239 10 4\n",
      "49\t13\t0.0\t3\t4 9\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLA_Fwd04-Rev25_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLA_Fwd04-Rev25_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLA_Fwd04-Rev25_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLA_Fwd04-Rev25_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLA_Fwd04-Rev25_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLA_Fwd04-Rev25_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLA_Fwd04-Rev25_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLA_Fwd04-Rev25_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        11,121 reads @  45.1 µs/read;   1.33 M reads/minute\n",
      "Finished in 0.511 s (45.928 µs/read; 1.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             11,121\n",
      "  Read 1 with adapter:                  11,117 (100.0%)\n",
      "  Read 2 with adapter:                  11,117 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        11,121 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,315,505 bp\n",
      "  Read 1:     3,294,318 bp\n",
      "  Read 2:     3,021,187 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      5,249,934 bp (83.1%)\n",
      "  Read 1:     2,750,638 bp\n",
      "  Read 2:     2,499,296 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 11117 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "41\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t0 0 1\n",
      "46\t9\t0.0\t3\t6 1 2\n",
      "47\t106\t0.0\t3\t69 21 16\n",
      "48\t821\t0.0\t3\t572 232 14 3\n",
      "49\t10154\t0.0\t3\t9726 413 15\n",
      "50\t24\t0.0\t3\t7 17\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 11117 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "36\t1\t0.0\t3\t1\n",
      "41\t1\t0.0\t3\t1\n",
      "42\t2\t0.0\t3\t2\n",
      "43\t2\t0.0\t3\t1 0 0 1\n",
      "44\t6\t0.0\t3\t3 1 1 1\n",
      "45\t45\t0.0\t3\t21 14 10\n",
      "46\t490\t0.0\t3\t230 247 13\n",
      "47\t10546\t0.0\t3\t10108 421 16 1\n",
      "48\t23\t0.0\t3\t16 7\n",
      "49\t1\t0.0\t3\t1\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLB_Fwd04-Rev27_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLB_Fwd04-Rev27_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLB_Fwd04-Rev27_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLB_Fwd04-Rev27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLB_Fwd04-Rev27_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLB_Fwd04-Rev27_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLB_Fwd04-Rev27_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLB_Fwd04-Rev27_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00         7,349 reads @  47.3 µs/read;   1.27 M reads/minute\n",
      "Finished in 0.361 s (49.110 µs/read; 1.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              7,349\n",
      "  Read 1 with adapter:                   7,346 (100.0%)\n",
      "  Read 2 with adapter:                   7,341 (99.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):         7,349 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,132,742 bp\n",
      "  Read 1:     2,158,990 bp\n",
      "  Read 2:     1,973,752 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      3,428,897 bp (83.0%)\n",
      "  Read 1:     1,799,721 bp\n",
      "  Read 2:     1,629,176 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 7346 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "44\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "46\t7\t0.0\t3\t3 2 1 1\n",
      "47\t55\t0.0\t3\t40 8 6 1\n",
      "48\t558\t0.0\t3\t381 174 3\n",
      "49\t6711\t0.0\t3\t6471 235 5\n",
      "50\t13\t0.0\t3\t7 6\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 7341 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "36\t1\t0.0\t3\t1\n",
      "42\t1\t0.0\t3\t0 1\n",
      "43\t3\t0.0\t3\t2 1\n",
      "44\t2\t0.0\t3\t0 1 0 1\n",
      "45\t35\t0.0\t3\t9 14 12\n",
      "46\t367\t0.0\t3\t215 146 4 2\n",
      "47\t6914\t0.0\t3\t6665 228 19 2\n",
      "48\t16\t0.0\t3\t10 6\n",
      "49\t2\t0.0\t3\t2\n",
      "\n",
      "Command: cutadapt --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLC_Fwd04-Rev28_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLC_Fwd04-Rev28_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLC_Fwd04-Rev28_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLC_Fwd04-Rev28_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.5 with Python 3.8.18\n",
      "Command line parameters: --cores 1 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLC_Fwd04-Rev28_L001_R1_001.fastq.gz -p /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-CasavaOneEightSingleLanePerSampleDirFmt-dlak_tqs/CTRLC_Fwd04-Rev28_L001_R2_001.fastq.gz --front AGGATTAGATACCCTGGTA -G CRRCACGAGCTGACGAC /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLC_Fwd04-Rev28_L001_R1_001.fastq.gz /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f28e9851-b0cc-49eb-95b3-0e8ee9c90fe4/data/CTRLC_Fwd04-Rev28_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:00:00        13,122 reads @  50.7 µs/read;   1.18 M reads/minute\n",
      "Finished in 0.675 s (51.421 µs/read; 1.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             13,122\n",
      "  Read 1 with adapter:                  13,122 (100.0%)\n",
      "  Read 2 with adapter:                  13,122 (100.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):        13,122 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,373,732 bp\n",
      "  Read 1:     3,851,377 bp\n",
      "  Read 2:     3,522,355 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      6,116,544 bp (83.0%)\n",
      "  Read 1:     3,209,659 bp\n",
      "  Read 2:     2,906,885 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGGATTAGATACCCTGGTA; Type: regular 5'; Length: 19; Trimmed: 13122 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "43\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "46\t19\t0.0\t3\t14 3 0 2\n",
      "47\t99\t0.0\t3\t62 21 16\n",
      "48\t1012\t0.0\t3\t705 298 7 2\n",
      "49\t11973\t0.0\t3\t11512 443 17 1\n",
      "50\t17\t0.0\t3\t4 12 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: CRRCACGAGCTGACGAC; Type: regular 5'; Length: 17; Trimmed: 13122 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-17 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t12\t12.8\t1\t0 12\n",
      "37\t1\t0.0\t3\t1\n",
      "41\t3\t0.0\t3\t2 1\n",
      "42\t4\t0.0\t3\t3 1\n",
      "43\t9\t0.0\t3\t6 3\n",
      "44\t7\t0.0\t3\t5 0 1 1\n",
      "45\t40\t0.0\t3\t14 12 12 2\n",
      "46\t594\t0.0\t3\t287 296 9 2\n",
      "47\t12433\t0.0\t3\t11954 458 18 3\n",
      "48\t19\t0.0\t3\t14 5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved SampleData[PairedEndSequencesWithQuality] to: analysis/seqs_trimmed/trimmed_sequences.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "#These sequences still have the primers attached - they need to be removed (using cutadapt) before denoising.\n",
    "\n",
    "!qiime cutadapt trim-paired \\\n",
    "--i-demultiplexed-sequences analysis/seqs/combined.qza \\\n",
    "--p-front-f AGGATTAGATACCCTGGTA \\\n",
    "--p-front-r CRRCACGAGCTGACGAC \\\n",
    "--p-error-rate 0.20 \\\n",
    "--output-dir analysis/seqs_trimmed \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba760d74",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir analysis/visualisations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c1d96380",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/trimmed_sequences.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "#Create a viewable summary file so the data quality can be checked. Viewing the quality plots generated here helps determine trim settings.\n",
    "!qiime demux summarize \\\n",
    "--i-data analysis/seqs_trimmed/trimmed_sequences.qza \\\n",
    "--o-visualization analysis/visualisations/trimmed_sequences.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c37a767d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/9d0d6f21-4397-4387-ad83-2b146eed1250')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 9d0d6f21-4397-4387-ad83-2b146eed1250>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load and display visualization for trimmed sequences\n",
    "path_to_trimmed_sequences_qzv = 'analysis/visualisations/trimmed_sequences.qzv'\n",
    "visualization_trimmed_sequences = Visualization.load(path_to_trimmed_sequences_qzv)\n",
    "display(visualization_trimmed_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4c5dae0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Denoising the data\n",
    "#Trimmed sequences are now quality assessed using the dada2 plugin within QIIME2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "cddef981",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada.R --input_directory /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/tmp23gn9uqa/forward --input_directory_reverse /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/tmp23gn9uqa/reverse --output_path /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/tmp23gn9uqa/output.tsv.biom --output_track /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/tmp23gn9uqa/track.tsv --filtered_directory /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/tmp23gn9uqa/filt_f --filtered_directory_reverse /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/tmp23gn9uqa/filt_r --truncation_length 180 --truncation_length_reverse 160 --trim_left 0 --trim_left_reverse 0 --max_expected_errors 2.0 --max_expected_errors_reverse 2.0 --truncation_quality_score 2 --min_overlap 12 --pooling_method independent --chimera_method consensus --min_parental_fold 1.0 --allow_one_off False --num_threads 0 --learn_min_reads 1000000\n",
      "\n",
      "Warning message:\n",
      "package ‘optparse’ was built under R version 4.2.3 \n",
      "R version 4.2.2 (2022-10-31) \n",
      "Loading required package: Rcpp\n",
      "\u001b[?25hDADA2: 1.26.0 / Rcpp: 1.0.11 / RcppParallel: 5.1.6 \n",
      "\u001b[?25h\u001b[?25h2) Filtering \u001b[?25h\u001b[?25h.......................\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "\u001b[?25h\u001b[?25h3) Learning Error Rates\n",
      "\u001b[?25h49439340 total bases in 274663 reads from 23 samples will be used for learning the error rates.\n",
      "43946080 total bases in 274663 reads from 23 samples will be used for learning the error rates.\n",
      "\u001b[?25h3) Denoise samples .......................\n",
      ".......................\n",
      "\u001b[?25h5) Remove chimeras (method = consensus)\n",
      "\u001b[?25h\u001b[?25h6) Report read numbers through the pipeline\n",
      "\u001b[?25h\u001b[?25h7) Write output\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[32mSaved FeatureTable[Frequency] to: analysis/dada2out/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: analysis/dada2out/representative_sequences.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[DADA2Stats] to: analysis/dada2out/denoising_stats.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime dada2 denoise-paired \\\n",
    "--i-demultiplexed-seqs analysis/seqs_trimmed/trimmed_sequences.qza \\\n",
    "--p-trunc-len-f 180 \\\n",
    "--p-trunc-len-r 160 \\\n",
    "--p-n-threads 0 \\\n",
    "--output-dir analysis/dada2out \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f4d49a3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Generate summary files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c4512f73",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/16s_denoising_stats.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime metadata tabulate \\\n",
    "--m-input-file analysis/dada2out/denoising_stats.qza \\\n",
    "--o-visualization analysis/visualisations/16s_denoising_stats.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "0d51106f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/4ea33aa8-f8b0-4c25-b431-da6d350381c5')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4ea33aa8-f8b0-4c25-b431-da6d350381c5>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_16s_denoising_stats_qzv = 'analysis/visualisations/16s_denoising_stats.qzv'\n",
    "visualization_16s_denoising_stats = Visualization.load(path_to_16s_denoising_stats_qzv)\n",
    "display(visualization_16s_denoising_stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "224b1ac7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/16s_table.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "--i-table analysis/dada2out/table.qza \\\n",
    "--m-sample-metadata-file data_files/metadata.tsv \\\n",
    "--o-visualization analysis/visualisations/16s_table.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c45b88fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/28a1e094-ef6c-4754-aaf8-abb79e06d665')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 28a1e094-ef6c-4754-aaf8-abb79e06d665>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_16s_table_qzv = 'analysis/visualisations/16s_table.qzv'\n",
    "visualization_16s_table = Visualization.load(path_to_16s_table_qzv)\n",
    "display(visualization_16s_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e57c00a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/16s_representative_seqs.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data analysis/dada2out/representative_sequences.qza \\\n",
    "--o-visualization analysis/visualisations/16s_representative_seqs.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "224b37c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/ea369f20-2a74-4e5e-b48a-dc0443eee9c9')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: ea369f20-2a74-4e5e-b48a-dc0443eee9c9>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_16s_representative_seqs_qzv = 'analysis/visualisations/16s_representative_seqs.qzv'\n",
    "visualization_16s_representative_seqs = Visualization.load(path_to_16s_representative_seqs_qzv)\n",
    "display(visualization_16s_representative_seqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90b54b45",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Section 2: Taxonomic Analysis¶\n",
    "#Assign taxonomy\n",
    "#classify each identical read or Amplicon Sequence Variant (ASV) to the highest resolution based on a database.\n",
    "#A classifier has already been trained for you for the V5V6 region of the bacterial 16S rRNA gene using the SILVA database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "9762f35c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Taxonomy] to: analysis/taxonomy/classification.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier silva_138_16s_v5v6_classifier_2021-4.qza \\\n",
    "--i-reads analysis/dada2out/representative_sequences.qza \\\n",
    "--p-n-jobs 1 \\\n",
    "--output-dir analysis/taxonomy \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "592ce449",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Generate a viewable summary file of the taxonomic assignments."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "be76c6a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/taxonomy.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime metadata tabulate \\\n",
    "--m-input-file analysis/taxonomy/classification.qza \\\n",
    "--o-visualization analysis/visualisations/taxonomy.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e5a5219",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Filtering\n",
    "#Generate a viewable summary file of the new table to see the effect of filtering."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "eb349b73",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: analysis/taxonomy/16s_table_filtered.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime taxa filter-table \\\n",
    "--i-table analysis/dada2out/table.qza \\\n",
    "--i-taxonomy analysis/taxonomy/classification.qza  \\\n",
    "--p-exclude Mitochondria,Chloroplast \\\n",
    "--o-filtered-table analysis/taxonomy/16s_table_filtered.qza \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "ef871e81",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/16s_table_filtered.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "--i-table analysis/taxonomy/16s_table_filtered.qza \\\n",
    "--m-sample-metadata-file data_files/metadata.tsv \\\n",
    "--o-visualization analysis/visualisations/16s_table_filtered.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "09a19fa2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/30e05cbe-9ba0-49fe-8ac8-35912600f9fd')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 30e05cbe-9ba0-49fe-8ac8-35912600f9fd>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_16s_table_filtered_qzv = 'analysis/visualisations/16s_table_filtered.qzv'\n",
    "visualization_16s_table_filtered = Visualization.load(path_to_16s_table_filtered_qzv)\n",
    "display(visualization_16s_table_filtered)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1f8f075",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Section 3: Build a phylogenetic tree¶"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "7a1b91ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: mafft --preservecase --inputorder --thread 1 /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/2aca41e4-4608-4f6b-8f17-bd396521c188/data/dna-sequences.fasta\n",
      "\n",
      "inputfile = orig\n",
      "700 x 289 - 240 d\n",
      "nthread = 1\n",
      "nthreadpair = 1\n",
      "nthreadtb = 1\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "  601 / 700 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "  690 / 700\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP   601 / 699 (thread    0)\n",
      "Reallocating..done. *alloclen = 1580\n",
      "\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "  600 / 700 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "  690 / 700\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP   601 / 699 (thread    0)\n",
      "Reallocating..done. *alloclen = 1591\n",
      "\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.520\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: FastTree -quote -nt /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/def188a9-7d0b-4dbc-907c-fae720fd3c1f/data/aligned-dna-sequences.fasta\n",
      "\n",
      "FastTree Version 2.1.11 Double precision (No SSE3)\n",
      "Alignment: /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/def188a9-7d0b-4dbc-907c-fae720fd3c1f/data/aligned-dna-sequences.fasta\n",
      "Nucleotide distances: Jukes-Cantor Joins: balanced Support: SH-like 1000\n",
      "Search: Normal +NNI +SPR (2 rounds range 10) +ML-NNI opt-each=1\n",
      "TopHits: 1.00*sqrtN close=default refresh=0.80\n",
      "ML Model: Jukes-Cantor, CAT approximation with 20 rate categories\n",
      "Initial topology in 0.29 seconds0 of    680   \n",
      "Refining topology: 38 rounds ME-NNIs, 2 rounds ME-SPRs, 19 rounds ML-NNIs\n",
      "Total branch-length 23.512 after 1.55 sec 1301 of 1364 nodes   \n",
      "ML-NNI round 1: LogLk = -35940.445 NNIs 136 max delta 16.94 Time 2.26ges (max delta 16.942)   \n",
      "Switched to using 20 rate categories (CAT approximation)7 of 20   \n",
      "Rate categories were divided by 1.146 so that average rate = 1.0\n",
      "CAT-based log-likelihoods may not be comparable across runs\n",
      "Use -gamma for approximate but comparable Gamma(20) log-likelihoods\n",
      "ML-NNI round 2: LogLk = -29636.544 NNIs 77 max delta 4.95 Time 2.77nges (max delta 4.947)   \n",
      "ML-NNI round 3: LogLk = -29585.729 NNIs 41 max delta 4.72 Time 3.08nges (max delta 4.720)   \n",
      "ML-NNI round 4: LogLk = -29564.812 NNIs 15 max delta 3.70 Time 3.27nges (max delta 3.700)   \n",
      "ML-NNI round 5: LogLk = -29561.566 NNIs 6 max delta 0.85 Time 3.37\n",
      "ML-NNI round 6: LogLk = -29561.532 NNIs 0 max delta 0.00 Time 3.40\n",
      "Turning off heuristics for final round of ML NNIs (converged)\n",
      "ML-NNI round 7: LogLk = -29536.380 NNIs 16 max delta 1.95 Time 3.99 (final)x delta 1.947)   \n",
      "Optimize all lengths: LogLk = -29535.828 Time 4.12 \n",
      "Total time: 4.92 seconds Unique: 683/700 Bad splits: 1/680 Worst delta-LogLk 0.185\n",
      "\u001b[32mSaved FeatureData[AlignedSequence] to: analysis/tree/aligned_16s_representative_seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[AlignedSequence] to: analysis/tree/masked_aligned_16s_representative_seqs.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Unrooted] to: analysis/tree/16s_unrooted_tree.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Rooted] to: analysis/tree/16s_rooted_tree.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "--i-sequences analysis/dada2out/representative_sequences.qza \\\n",
    "--o-alignment analysis/tree/aligned_16s_representative_seqs.qza \\\n",
    "--o-masked-alignment analysis/tree/masked_aligned_16s_representative_seqs.qza \\\n",
    "--o-tree analysis/tree/16s_unrooted_tree.qza \\\n",
    "--o-rooted-tree analysis/tree/16s_rooted_tree.qza \\\n",
    "--p-n-threads 1 \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "350f7064",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Section 4: Basic visualisations and statistics\n",
    "#ASV relative abundance bar charts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7009ff1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create bar charts to compare the relative abundance of ASVs across samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "f0a936c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/barchart.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime taxa barplot \\\n",
    "--i-table analysis/taxonomy/16s_table_filtered.qza \\\n",
    "--i-taxonomy analysis/taxonomy/classification.qza \\\n",
    "--m-metadata-file data_files/metadata.tsv \\\n",
    "--o-visualization analysis/visualisations/barchart.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "2416e8aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f12921fc-62bd-4ca5-9cb9-cce17ad666e2')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f12921fc-62bd-4ca5-9cb9-cce17ad666e2>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_barchart_qzv = 'analysis/visualisations/barchart.qzv'\n",
    "visualization_barchart = Visualization.load(path_to_barchart_qzv)\n",
    "display(visualization_barchart)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "64e5900b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/84666e8e-d24b-469c-9b40-21f24e6ff11f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-pbqv10gq\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/cde26a3e-433f-4820-a6b5-bb472ab2d611/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-zpl2j0qc\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/2322f492-5d61-4248-9d55-36834ea416ed/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-4d8bh1ds\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/1b49351a-12ae-4e4f-a9bf-e54f58b463bc/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-9qi8gxdr\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/60088f34-25c7-4b13-a805-ce471a90e404/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-bk5y01gs\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/03a9bf68-e294-4261-9c75-46144d968fb3/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-uvf8amom\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/52d6f9d3-613a-40f1-ab2d-0cf16312c3d5/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-pu9lo82h\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/43d1247c-153b-4a4e-bdcf-caa4a0856761/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-4o367avn\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/36070a73-be5d-449a-8c6e-3cc8720a6671/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-9dd5sq99\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/73bdc7cf-dc34-4b3c-9d84-e30987b73c69/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-sik_j07q\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/90df2e15-e60a-4c86-b99f-a8d20b3f9b0f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-urmdvjs8\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/e7100321-5704-479a-8db1-3f727efc521e/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-n42vcp89\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/dde52d67-77b1-412e-a20a-bd246b0448df/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-w1h9zr1c\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f54652b3-24d5-49d5-9f36-f2661bcec89b/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-2rcyj8iv\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/dbd9ae3f-d4c7-4d55-9fbb-26bcf74d8f11/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-tj_5g1kd\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/2a1e3fd3-c2a0-4a93-ac0b-8da9601a2d26/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-2jg669qx\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/28fda918-d60e-4ea5-940a-31b55854e3b8/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-9qfsu0mr\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/bbd2df34-bc1e-46ae-8e7c-aa19f0d52219/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-rlmlvzym\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/0f77691c-ec06-46a6-9067-192e1102fb1b/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-f97na0vc\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f89200e1-6bd7-420a-9eda-51fc91088b31/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-hdes796h\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/a0ae3f89-4e5e-4a22-994e-006f94ded3c9/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-_bhjfsso\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/0473a7cf-2919-43f3-a092-433fed16839a/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-ary527m_\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/a4359223-814e-4718-bc67-b16279e10922/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-m4t17f5g\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/394678d9-0878-4ad5-8200-bfa4ab0e1d69/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-9boyyfyo\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/5023d5db-16bb-4b38-8f2c-92350b39988f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-vx1l3zvu\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/af625abb-674c-4a5e-90b5-d62ae9f5a8ff/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-8xko4rz2\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/5852c85a-36f4-41d6-8726-7e397f3be564/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-ny8g0s2q\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f7ae4058-b5a2-46d7-bb4f-21b128749c7f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-_e3fvp8g\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/e97d9045-e97c-46b1-8e12-b3e8da585144/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-c9kbez6m\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/dc41378e-3a97-4c7c-828a-8429ea15d51f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-iqfp9dl3\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/67457d88-4f7c-4b9e-9df2-673617b552f9/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-2_vn7k01\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/a1c54bfa-a313-4e3a-b526-ada0de3dcd29/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-_ldi3dp2\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/95313a43-14a1-489d-8cd8-bb2d1bbc388c/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-n30tvy3h\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/6572a1ad-e5ac-4c46-be6d-565f7c93274c/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-hnx6_bma\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/d4bf8acc-2fb1-4962-92aa-e83ec624af74/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-i0kwgbu4\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f5ca195e-69e2-42d2-a8a2-9607b727f0ff/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-mkl43820\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/1fad532f-91a0-41b2-bc37-0050b601e493/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-0t189ahw\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/4d928ab3-b001-476c-82c7-d3f26ba1e422/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-38vmgijr\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/ca1bcb8e-2153-4fe2-8bd8-03345259e5e5/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-g8fmg5ja\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/a263fcf4-f715-45f9-8621-b5f6f723df72/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-zcue4_wp\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/8711d0db-cd0b-4c9d-82b3-57e629638f8a/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-v9_vath7\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/337cfb32-b218-484f-80dd-af9d0214f848/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-b2q2el8z\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/67a87507-95b6-4c24-a918-fc60c75238fb/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-qzd1hisb\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/032cc766-51f3-45e2-84ca-1d7b587af4c7/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-mjdp0oz1\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/22dd17a0-d4b5-4b07-a012-002af61bf636/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-6wgt53lx\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/8a9ed4b7-4383-43be-8ce6-461a988d0354/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-1sdf6kbb\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/ab1c8548-422e-4371-94b5-0ea5163e3901/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-7y_m6hnm\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/8d1a66c2-ef26-4b24-9866-f34b6e861001/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-1kwoy4xh\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/61defbe0-24ac-4cb6-a5c3-718047b922d2/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-h168181b\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/c92fff23-0ded-4933-ba01-3386d94d5937/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-1ia9okni\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/00dc3719-8980-460c-894f-89d0ae41a985/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-iehvdjl1\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/0401f53d-8815-43cf-a86b-0bd176ea1b6f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-jjg0xgrr\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/01b75fce-d0cd-4720-8a7e-c5f62f77ec89/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-152m11x9\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/927fb992-8387-4990-8cad-eb38259a88e8/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-4fjp_lv3\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/910edcbe-12d0-4c99-8b86-602d0cf5a7cc/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-b41mkurn\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/2f4e7072-e8eb-4664-96a6-0ebd57d627f1/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-5u8obxko\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/ce41a8f0-e3a0-4565-a3d7-3e07248d4eb8/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-_f3vk7f8\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/51fcbf17-6a22-459e-a8f9-ba97af2672e7/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-tj9g16zg\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/c80e35b0-c1e4-4046-a46a-2f7f620b8e67/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-jb11tf_0\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/aaa3aed9-f9f9-45f2-b631-a79ccbab6317/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-oqu1u8_z\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/06bee33f-aa18-45f0-9c0a-316b37d83d48/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-37aky3rx\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/528fab54-1d96-4fd1-940c-d8a5539c4f86/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-6f3t_pyi\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/cb6ac339-aae4-45fb-8833-34cad5cd9408/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-z9cyrj6o\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/46068a12-4780-471f-a9b1-3c06d8b1d5d8/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-60tay5m0\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/b3d45889-cd85-4b97-9a5f-1aeec2cd8e47/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-z3qw5d18\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/039422a3-52ea-46f2-95b6-6645c290151a/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-g3ziy512\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/e2831cfd-4727-4b8e-94db-3af6257936f7/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-y_ck9u9i\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7a448c35-ab52-429e-b880-da1bdbedbf4e/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-6g9rg926\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/cc370df7-8ee2-4f97-a36d-a8f6b502c2a0/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-f0ipr7iq\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/2f82e207-bf7a-4d79-a549-3b81db82e800/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-3vv89nmv\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/b298ac22-0c26-413a-88fe-9261a10133ef/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-css1bo0m\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7683c178-8baa-4d01-87d7-bdc716f14ef4/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-v0dt3110\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/2f240a3e-79f4-4600-b0f2-aad88d05c1b3/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-zdy4zdnz\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/96e9318c-5df9-4e71-9ce4-16158e59c771/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-3q9xtip4\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/817ef61f-3550-443f-ad71-ff8358ce30fc/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-vpuy1hkn\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/6f135ed8-c173-4c3b-ba40-f31c7c3ed2a5/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-0hc0xuxe\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/21b46fa1-8320-4c50-8e40-e85c48e07aa6/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-es9zd6qp\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/473cf244-32d4-489e-b0ef-340cfc4ae5cf/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-io5xr2av\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/6c85f88d-f75f-471a-93bd-dcf1f5ca43c1/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-oxzhkk87\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/04674671-cf6e-4fca-ad5c-b336d03ebad5/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-o74gp70x\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/665cd0ea-7367-4fe8-a570-ca04f4381ee6/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-691bc5kw\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/121911de-f58d-44f5-8659-5f9f9e176ce4/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-479p7fnm\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/653eb2a1-e3ae-4650-b7be-cefd9b6e77bb/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-b3qe98yp\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/271275d3-214c-453c-b0c7-090605f73501/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-i3e0xdnn\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/759e00db-94bd-465d-a1f5-13eda33c1ba9/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-s5u2hlgp\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/9a9a7af9-7601-487e-9f07-d6d1ffc26f6c/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-xaiyfj_f\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/f1de1d8e-9e0d-42b7-b515-03ee1cfcd53f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-6x7l48w2\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/35fada1c-3f94-4559-9426-b92e94506097/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-h9bnvify\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/5efb4228-f148-4b27-bf97-6e26e8fff099/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-jfmu23d9\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/5deae329-bf8c-4e3c-8da2-20aa76b64131/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-k8nffz9v\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/4c240695-3cf5-461c-9f62-ae89afc3b18d/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-ubb4lzs7\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/d148e73f-a1eb-433a-8643-fdde5bbc937f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-1_do1631\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/fdc2ac07-4ccc-49e5-90e4-c5898e55fc9e/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-lvqepyo9\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/0d9eb3b5-df5e-46a2-a3c5-8202a74bfde2/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-ni51l9z3\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/0974c000-7f99-42d8-bf60-7ee3f29d3772/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-v10jmr7h\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c1115ed-47c6-4be7-8451-c97c6c8c0cb9/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-k8hku985\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/fd1578b6-8ade-444d-a287-1e8f1f7faead/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-gymx9tdo\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/18e985b6-fa94-4268-a23b-2e8141df2c1f/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-vrtqi4m1\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/85955b7d-5d72-48d0-830c-e50048f884ca/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-il64mjgo\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command:\n",
      "\n",
      "faithpd -i /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/0d912e7e-aa9d-4020-ac07-90b54834015e/data/feature-table.biom -t /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/7c7ad574-3873-4de6-aa66-64895eca559e/data/tree.nwk -o /var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/q2-AlphaDiversityFormat-oyzam113\n",
      "\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "/Users/laurenstoeltje/miniconda3/envs/qiime2-amplicon-2023.9/lib/python3.8/site-packages/q2_diversity/_alpha/_visualizer.py:245: FutureWarning: The default value of numeric_only in DataFrameGroupBy.median is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.\n",
      "  median_ = grouped.median()\n",
      "\u001b[32mSaved Visualization to: analysis/visualisations/16s_alpha_rarefaction.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "#Rarefaction curves\n",
    "\n",
    "!qiime diversity alpha-rarefaction \\\n",
    "--i-table analysis/taxonomy/16s_table_filtered.qza \\\n",
    "--i-phylogeny analysis/tree/16s_rooted_tree.qza \\\n",
    "--p-max-depth 9062 \\\n",
    "--m-metadata-file data_files/metadata.tsv \\\n",
    "--o-visualization analysis/visualisations/16s_alpha_rarefaction.qzv \\\n",
    "--verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "27d6b473",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/e80ee534-aabd-4149-9154-a8df6d555741')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: e80ee534-aabd-4149-9154-a8df6d555741>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_qzv = '/Users/laurenstoeltje/final_project_668/analysis/visualisations/16s_alpha_rarefaction.qzv'\n",
    "visualization = Visualization.load(path_to_qzv)\n",
    "display(visualization)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df809698",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Alpha and beta diversity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "253b236e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: analysis/diversity_metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: analysis/diversity_metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: analysis/diversity_metrics/observed_features_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: analysis/diversity_metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: analysis/diversity_metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: analysis/diversity_metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: analysis/diversity_metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: analysis/diversity_metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: analysis/diversity_metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: analysis/diversity_metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: analysis/diversity_metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: analysis/diversity_metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: analysis/diversity_metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: analysis/diversity_metrics/unweighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: analysis/diversity_metrics/weighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: analysis/diversity_metrics/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: analysis/diversity_metrics/bray_curtis_emperor.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny analysis/tree/16s_rooted_tree.qza \\\n",
    "  --i-table analysis/taxonomy/16s_table_filtered.qza \\\n",
    "  --p-sampling-depth 5583 \\\n",
    "  --m-metadata-file data_files/metadata.tsv \\\n",
    "  --output-dir analysis/diversity_metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61ddc125",
   "metadata": {},
   "outputs": [],
   "source": [
    "cp analysis/diversity_metrics/*.qzv analysis/visualisations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6acd01f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Faith Phylogenetic Diversity (a measure of community richness) and evenness metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "18a7e26b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/faith-pd-group-significance.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity analysis/diversity_metrics/faith_pd_vector.qza \\\n",
    "  --m-metadata-file data_files/metadata.tsv \\\n",
    "  --o-visualization analysis/visualisations/faith-pd-group-significance.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c9a222a2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/762c0c8f-e51a-45d7-9b5a-6f3300f48441')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 762c0c8f-e51a-45d7-9b5a-6f3300f48441>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_faith_pd = 'analysis/visualisations/faith-pd-group-significance.qzv'\n",
    "visualization_faith_pd = Visualization.load(path_to_faith_pd)\n",
    "display(visualization_faith_pd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "838a2d45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/evenness-group-significance.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity analysis/diversity_metrics/evenness_vector.qza \\\n",
    "  --m-metadata-file data_files/metadata.tsv \\\n",
    "  --o-visualization analysis/visualisations/evenness-group-significance.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "7f122d2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/ad7bae4f-9e1b-460c-b2b0-6c14415d2461')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: ad7bae4f-9e1b-460c-b2b0-6c14415d2461>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_evenness = 'analysis/visualisations/evenness-group-significance.qzv'\n",
    "visualization_evenness = Visualization.load(path_to_evenness)\n",
    "display(visualization_evenness)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f76b475",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PERMANOVA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "1ee6bbf5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/55a7574f-b379-46dc-a02a-bb2ed64bb78c')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 55a7574f-b379-46dc-a02a-bb2ed64bb78c>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_unweighted_unifrac = 'analysis/visualisations/unweighted_unifrac_emperor.qzv'\n",
    "visualization_unweighted_unifrac = Visualization.load(path_to_unweighted_unifrac)\n",
    "display(visualization_unweighted_unifrac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "092a6425",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: analysis/visualisations/unweighted-unifrac-genotype-significance.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix analysis/diversity_metrics/unweighted_unifrac_distance_matrix.qza \\\n",
    "  --m-metadata-file data_files/metadata.tsv \\\n",
    "  --m-metadata-column Genotype \\\n",
    "  --o-visualization analysis/visualisations/unweighted-unifrac-genotype-significance.qzv \\\n",
    "  --p-pairwise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "62b6e8be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/d5e95f72-e2cf-40a4-8076-e0beea5185ed')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: d5e95f72-e2cf-40a4-8076-e0beea5185ed>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_unweighted_unifrac_genotype = 'analysis/visualisations/unweighted-unifrac-genotype-significance.qzv'\n",
    "visualization_unweighted_unifrac = Visualization.load(path_to_unweighted_unifrac_genotype)\n",
    "display(visualization_unweighted_unifrac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd99996b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Unweighted UniFrac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "759a0b31",
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix analysis/diversity_metrics/unweighted_unifrac_distance_matrix.qza \\\n",
    "  --m-metadata-file data_files/metadata.tsv \\\n",
    "  --m-metadata-column Environment \\\n",
    "  --o-visualization analysis/visualisations/unweighted-unifrac-environment-significance.qzv \\\n",
    "  --p-pairwise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "bc36da54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/cb/bxr92qgn10d978brzl0pdtfc0000gn/T/qiime2/laurenstoeltje/data/4a218fe3-4ec1-45ad-b615-ea8d95760e43')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4a218fe3-4ec1-45ad-b615-ea8d95760e43>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_to_unweighted_unifrac_env = 'analysis/visualisations/unweighted-unifrac-environment-significance.qzv'\n",
    "visualization_unweighted_unifrac = Visualization.load(path_to_unweighted_unifrac_env)\n",
    "display(visualization_unweighted_unifrac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5ce929d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Results:\n",
    "#Substantial decreases in bacterial alpha diversity were observed in all “sterile” anemone genotypes.\n",
    "#Simpson’s and Shannon’s bacterial diversity also decreased in all anemones, regardless of genotype, with significant reductions in AIMS2 (Shannon’s only), AIMS3 (both Shannon’s and Simpson’s), and AIMS4 (Shannon’s only).\n",
    "#A total of 13 bacterial families were found in at least three genotypes of control anemones and absent in at least three genotypes of “sterile” anemones.\n",
    "#Conversely, there were five families that were present in the “sterile” anemones, yet absent in control anemones.\n",
    "#PERMANOVA results by genotype confirmed that the bacterial community structure varied significantly with the interaction of sample type and environment.\n",
    "\n",
    "#Analysis of these results suggest that the bacterial diversity and community structure for each of four anemone genotypes was significantly different with a reduction in alpha diversity when anemones were moved from unfiltered seawater into sterile seawater."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
